peopleforce/demo-blog-app
# [Test Application] Add import of comments

### Description

**NOTE:** Please take a look at the changes through commit history as there is first additional commit for a full rails app

What the changes are:

- Added the full rails setup.
- Changed logic inside `Comment` model due to `Faker::UniqueGenerator::RetryLimitExceeded`
- Current changes implemented the logic of importing rows from `comments.csv` file to the Rails application by associated `Post` as the new instances of `Comment` model.
- Added logging of the validation errors and logging for the steps of script running.
- Validation errors do not break script run, I suggest the data inside csv could be changed according to the validation.

### How to test locally
Checkout this branch

- Please ensure `bundle exec rake db:migrate`, `bundle exec rake db:seed` have run
- Run `bundle exec script/comments/import`
- Ensure there are not errors while script is running, with logging output. For example:
<img width="892" alt="image" src="https://github.com/peopleforce/demo-blog-app/assets/79387360/fd86d67d-136d-4976-b02a-32d6c3456a12">

- Run `bundle exec rails console`
- Ensure `Comment.all.size` returns the same number as `Count of created comments: <number>`

```
irb(main):015> Comment.all.size
  Comment Count (0.2ms)  SELECT COUNT(*) FROM "comments"
=> 1273
```

### Risk
**None** Only dev script has been added
